### PR TITLE
GSoC proposal for ACTS GPU R&D

### PR DESCRIPTION
**Pull request by beomki-yeo**

### CODE REVIEW

File: _gsocproposals/2022/proposal_ACTS_GPU_pipeline_optimization.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+## Task Ideas
+
+* Benchmarking downstream caching allocators provided by [vecmem][vecmem] for [traccc][traccc] algorithms
+* Benchmarking the unified memory and device memory for CUDA memory allocation
+* Application of CUDA-MPS and MIG to the pipeline
+
+## Expected Results
+
+We are expected to observe the actual acceleration of GPU pipeline, especially for CUDA API.
```
Comment (cgleggett) [inline _gsocproposals/2022/proposal_ACTS_GPU_pipeline_optimization.md:33]: increased throughput

File: _gsocproposals/2022/proposal_ACTS_GPU_pipeline_optimization.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
```
Comment (agheata) [inline _gsocproposals/2022/proposal_ACTS_GPU_pipeline_optimization.md:33]: This is very minimalistic as expected results. Please expand with more clear/measurable objectives. What kind of acceleration is expected? Benchmarking which use case? We are also encouraging stating here also what the student is expected to learn/achieve in this process.

File: _gsocproposals/2022/proposal_ACTS_GPU_pipeline_optimization.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+mentor_avail: June-August
+organization:
+  - LBNL
+---
+
+## Description
+
+A Common Tracking Software ([Acts][Acts]) is a general tracking software toolkit for High Energy Physics experiments. 
+The Acts collaboration has launched several R&D lines ([vecmem][vecmem], [detray][detray], and [traccc][traccc]) 
+for GPU acceleration by parallelizing the track reconstruction. vecmem is a memory management toolkit which provides
```
Comment (agheata) [inline _gsocproposals/2022/proposal_ACTS_GPU_pipeline_optimization.md:17]: For readability, maybe emphasize or use quotes for `vecmem`, `detray` and `traccc` when they start a sentence.

**Review by agheata — CHANGES_REQUESTED** [review]

File: _gsocproposals/2022/proposal_ACTS_GPU_pipeline_optimization.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+## Requirements
+
+ * Strong C++ skills
+ * Experience with GPU APIs (CUDA, SYCL, etc.)
+ * Linux, Git, and Shell scripting
+
+## Mentors
+
+ * **[Beomki Yeo](mailto:beomki.yeo@berkeley.edu)** (LBNL)
```
[the remark this answers is not in the document]
Comment (beomki-yeo) [reply _gsocproposals/2022/proposal_ACTS_GPU_pipeline_optimization.md:49]: I added @cgleggett

**Review by agheata — APPROVED** [review]

File: _gsocproposals/2022/proposal_ACTS_GPU_pipeline_optimization.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+## Requirements
+
+ * Strong C++ skills
+ * Linux, Git, and Shell
+ * Experience with any of GPU APIs (CUDA, SYCL, etc.) will be advantageous
+
+## Mentors
+
+ * **[Beomki Yeo](mailto:beomki.yeo@berkeley.edu)** (LBNL)
+ * [Charles Leggett](cgleggett@lbl.gov) (LBNL)
```
Comment (agheata) [inline _gsocproposals/2022/proposal_ACTS_GPU_pipeline_optimization.md:44]:
```suggestion
 * [Charles Leggett](mailto:cgleggett@lbl.gov) (LBNL)
```

File: gsoc/2022/mentors.md [hunk 6]
```diff
@@ -28,6 +28,7 @@ layout: plain
 * Mario Lassnig [mario.lassnig@cern.ch](mailto:mario.lassnig@cern.ch) CERN
 * David Lange [david.lange@cern.ch](mailto:david.lange@cern.ch) Princeton
 * Wim Lavrijsen [wlavrijsen@lbl.gov] LBL
+* Charles Leggett [cgleggett@lbl.gov] LBNL
```
Comment (agheata) [inline gsoc/2022/mentors.md:31]:
```suggestion
* Charles Leggett [cgleggett@lbl.gov](mailto:cgleggett@lbl.gov) LBNL
```

**Review by agheata — APPROVED** [review]

**Pull request merged**